### PR TITLE
Sync failed dialog

### PR DESCRIPTION
**Pull request by sanchita**
Pivotol story -> https://www.pivotaltracker.com/story/show/164243208

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/sync/indicator/dialog/SyncIndicatorFailureDialog.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.fragment.app.FragmentManager
+import org.simple.clinic.R
+
+class SyncIndicatorFailureDialog : AppCompatDialogFragment() {
+
+  companion object {
+    private const val KEY_MESSAGE = "failure_message"
+
+    fun show(fragmentManager: FragmentManager, message: String) {
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/sync/indicator/dialog/SyncIndicatorFailureDialog.kt]: Will `message` be used in a future commit?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -333,9 +333,14 @@
...
+
+  <!--- Sync Indicator-->
   <string name="sync_indicator_status_pending">Sync pending</string>
   <string name="sync_indicator_status_failed">Sync failed</string>
   <string name="sync_indicator_status_syncing">Syncing…</string>
   <string name="sync_indicator_status_synced_just_now">Synced just now</string>
   <string name="sync_indicator_status_synced">Synced %d min ago</string>
+  <string name="sync_indicator_dialog_error_network">Make sure you have a working internet connection to sync patient records.</string>
+  <string name="sync_indocator_dialog_button_text">okay</string>
+  <string name="sync_indicator_dialog_error_server">Sync servers are down at the moment. Please try again later or get in touch with your CVHO.</string>
```
Comment (saket) [inline app/src/main/res/values/strings.xml]: We generally skip spaces for prefixing resource IDs. E.g.: `bloodpressureentry`, `patientsummary`, etc. Can we fix these strings as well?

File: app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt [hunk 3]
```diff
@@ -80,10 +80,8 @@ class SyncIndicatorViewController @Inject constructor(
     return showSyncIndicatorState.mergeWith(showDefaultSyncIndicatorState)
   }
 
-  private fun syncIndicatorState(timestamp: Instant?, maxIntervalSinceLastSync: Duration): SyncIndicatorState {
-    if (timestamp == null) {
-      return SyncPending
-    }
+  private fun syncIndicatorState(syncState: LastSyncedState, maxIntervalSinceLastSync: Duration): SyncIndicatorState {
+    val timestamp = syncState.lastSyncSucceededAt ?: return SyncPending
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt]: Nice!

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (sanchita) [reply app/src/main/res/values/strings.xml]: Sure.

File: app/src/main/java/org/simple/clinic/sync/indicator/dialog/SyncIndicatorFailureDialog.kt (hunk 1 above)
Comment (sanchita) [reply app/src/main/java/org/simple/clinic/sync/indicator/dialog/SyncIndicatorFailureDialog.kt]: Oh no. Good catch.
I have to pass this message :P

**Pull request merged**